font-codegen/src/record.rs

```diff
@@ -273,6 +273,7 @@ pub(crate) fn generate_compile_impl(
         let default_field_inits = fields.iter_compile_default_inits();
         quote! {
         impl #default_impl_params Default for #name <#generic_param> {
+            #[allow(clippy::derivable_impls)]
```

Comment:
You'll notice above that this path is only taken if we believe that we can't derive the impl, otherwise we just derive it, so perhaps you're hitting some case where `can_derive_default` is returning `false` incorrectly?

Reply:
Yes. If you `#[default(the default value for the type)]`, e.g. `#[default(0)]` and all other fields are defaultable we don't identify that we could just take the derived default. We generate roughly:

```rust
fn default() -> Self {
Self {
  my_int_field_with_default_zero: 0,
  other_fields: Default::default()
}
```

 It's a bug in codegen but not one I want to block on. Your other change will avoid it but the bug will remain, will file an issue.

Reply:
https://github.com/googlefonts/fontations/issues/1376